frontend/taipy-gui/src/hooks/useLocalStorageWithEvent.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2021-2024 Avaiga Private Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+import { Dispatch, useEffect } from "react";
+import { createLocalStorageAction, createLocalStorageUpdateAction, TaipyBaseAction } from "../context/taipyReducers";
+
+const STORAGE_EVENT = "storage";
+const CUSTOM_LOCAL_STORAGE_EVENT = "local-storage";
+
+export const useLocalStorageWithEvent = (dispatch: Dispatch<TaipyBaseAction>) => {
+    // Override the original setItem and removeItem behaviour for localStorage to dispatch a custom storage event for local tab
+    useEffect(() => {
+        // Preserve the original setItem and removeItem method
+        const _setItem = Storage.prototype.setItem;
+        const _removeItem = Storage.prototype.removeItem;
+
+        Storage.prototype.setItem = function (key, value) {
```

Comment:
a bit invasive ...
Any other way ?

Reply:
I have tried multiple ways but to be able to catch an event on the local tab, I still need to modify the original implementation